### DISCUSSION TITLE
Encapsulation

### DISCUSSION THREAD

**Discussion by gryznar** [modular/modular#430] [Q&A]
Somewhere I have heard / seen that Mojo will have true encapsulation in the opposite to Python in which encapsulation does not exist (only convention). 
I am just wondering.. Will this encapsulation be as much powerful as e.g. in C++ in which also inheritance may be controlled?

**Comment by mojodojodev** [comment]
I think you're talking about how Python just has `__` for member variables that name mangles but can still be accessed, or `_` that is just a convention for private members.

In Mojo it's not enforced no, but in a similar question Chris said that they may do this in the future: https://github.com/modularml/mojo/discussions/129#discussioncomment-5892035

**Reply by gryznar** [reply]
Yes, exactly about that. I hope that true encapsulation will be implemented in Mojo!
